test/transcoding/SPV_INTEL_non_constant_addrspace_printf/non-constant-printf.ll

```diff
@@ -0,0 +1,79 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: not llvm-spirv %t.bc -o %t.spv 2>&1 | FileCheck %s --check-prefix=CHECK-WO-EXT
+
+; RUN: llvm-spirv -spirv-text %t.bc -o %t.spt --spirv-ext=+SPV_INTEL_non_constant_addrspace_printf
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv -to-binary %t.spt -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis %t.rev.bc -o %t.rev.ll
+; RUN: FileCheck < %t.rev.ll %s --check-prefix=CHECK-LLVM
+
+; CHECK-WO-EXT: RequiresExtension: Feature requires the following SPIR-V extension:
+; CHECK-WO-EXT: The SPV_INTEL_non_constant_addrspace_printf extension should be allowed to translate this module, because this LLVM module contains the printf function with format string, whose address space is not equal to 2 (constant).
+
+; CHECK-SPIRV: Capability NonConstantAddrspacePrintfINTEL
+; CHECK-SPIRV: Extension "SPV_INTEL_non_constant_addrspace_printf"
+; CHECK-SPIRV: ExtInstImport [[#ExtInstSetId:]] "OpenCL.std"
+; CHECK-SPIRV: TypeInt [[#TypeInt8Id:]] 8 0
+; CHECK-SPIRV: TypeInt [[#TypeInt32Id:]] 32 0
+; CHECK-SPIRV: Constant [[#TypeInt32Id]] [[#]] 10
+; CHECK-SPIRV: Constant [[#TypeInt32Id]] [[#PrintfAdditionalArg:]] 1
+; CHECK-SPIRV: TypeArray [[#TypeArray1Id:]] [[#TypeInt8Id]] 10
+; CHECK-SPIRV: TypePointer [[#Ptr1:]] 5 [[#TypeArray1Id]]
+; CHECK-SPIRV: TypeArray [[#TypeArray2Id:]] 2 19
+; CHECK-SPIRV: TypePointer [[#Ptr2:]] 5 [[#TypeArray2Id]]
+; CHECK-SPIRV: TypePointer [[#Ptr3:]] 5 [[#TypeInt8Id]]
+; CHECK-SPIRV: Variable [[#Ptr1]] [[#Var1:]] 5 12
+; CHECK-SPIRV: Variable [[#Ptr2]] [[#Var2:]] 5 21
+; CHECK-SPIRV: InBoundsPtrAccessChain [[#Ptr3]] [[#PrintfFormat1:]] [[#Var1]]
+; CHECK-SPIRV: ExtInst [[#TypeInt32Id]] [[#]] [[#ExtInstSetId]] printf [[#PrintfFormat1]]
+; CHECK-SPIRV: InBoundsPtrAccessChain [[#Ptr3]] [[#PrintfFormat2:]] [[#Var2]]
+; CHECK-SPIRV: ExtInst [[#TypeInt32Id]] [[#]] [[#ExtInstSetId]] printf [[#PrintfFormat2]] [[#PrintfAdditionalArg]]
+
+; CHECK-LLVM: internal unnamed_addr addrspace(1) constant [6 x i8] c"Test\0A\00", align 1
+; CHECK-LLVM: internal unnamed_addr addrspace(1) constant [10 x i8] c"Test: %d\0A\00", align 1
+; CHECK-LLVM: getelementptr inbounds [6 x i8], [6 x i8] addrspace(1)* @0, i32 0, i32 0
+; CHECK-LLVM: call spir_func i32 (i8 addrspace(1)*, ...) @printf(i8 addrspace(1)* %1)
```

Comment:
Why addrspace has changed from 5 to 1 after translation?

Reply:
Because https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/lib/SPIRV/SPIRVInternal.h#L214 StorageClassCrossWorkgroup is mapped to SPIRAS_Global. Should it be translated in another way?

Reply:
Why do we have storage class 5 in SPIR-V? According to the map you mentioned `addrspace(5)` (which is `SPIRAS_GlobalDevice`) should be translated from LLVM to SPIR-V as  `StorageClassDeviceOnlyINTEL`, which is [5936](https://github.com/KhronosGroup/glslang/blob/c34bb3b6c55f6ab084124ad964be95a699700d34/SPIRV/spirv.hpp#L215), right?

Yes, it should be translated in another way. I'd also recommend to choose other addrspace for the tests, i.e. in range from 0 to 4, and not 2 - it is constant address space.

Reply:
If USM storage spaces extension is disabled during the translation - globa_device/host_as is being silently replaced with global_as. I'd also recommend to pick another AS for this test.

Reply:
Thanks, got it